client/types.go

```diff
@@ -19,8 +19,25 @@ var DefaultClient = &Client{
 	healthCheck: defaultHealthCheck,
 }
 
+type SDKClient interface {
+	client.APIClient
+	// TODO declare SDK higher-order functions
```

Comment:
```suggestion
	// TODO declare SDK higher-order functions
	DaemonHostWithContext(ctx context.Context)
```

Reply:
`DaemonHostWithContext` is required for compatibility with the SDK existing methods, but I wonder we want this exposed to the users and I'm not sure about the use-case for this API method override.

What I would actually expect here is `Pull` (with JSON message decoding),  `Run` and `Exec` (with attach and stream demultiplexing support), etc...